src/core/obj.js

```diff
@@ -1147,8 +1148,33 @@ var XRef = (function XRefClosure() {
               uncompressed: true,
             };
           }
-          var contentLength = skipUntil(buffer, position, endobjBytes) + 7;
-          var content = buffer.subarray(position, position + contentLength);
+          let contentLength, startPos = position + token.length;
+
+          // Find the next "obj" string, rather than "endobj", to ensure that
+          // we won't skip over a new 'obj' operator in corrupt files where
+          // 'endobj' operators are missing (fixes issue9105_reduced.pdf).
+          while (startPos < buffer.length) {
+            let endPos = startPos + skipUntil(buffer, startPos, objBytes) + 4;
+            contentLength = endPos - position;
+
+            let tokenStr = bytesToString(buffer.subarray(endPos - 8, endPos));
+            if (endobjRegExp.test(tokenStr)) {
+              break; // Found 'endobj'.
+            } else {
+              let objStartPos = Math.max(endPos - 25, startPos);
```

Comment:
What does this number 25 stand for? Can we make a constant out of it or add a comment?

Reply:
Since a `1 0 obj` (or similar) sequence can contain arbitrary whitespace, we need to check enough of the data to ensure that we won't miss e.g. `1     0     obj`.

In this case though, I'd agree that a constant is probably better especially considering that this number is somewhat randomly chosen.